draft-pauly-add-deer.md

```diff
@@ -65,8 +84,256 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+using other encrypted transports.
+
+#### SVCB Records for Equivalent Encrypted Resolvers using DoH
+
+The following example shows an Equivalent Encrypted Resolver using DoH, as returned by a query
+for an SVCB record for "dns://resolver.arpa":
+
+~~~
+   _dns.resolver.arpa  7200  IN SVCB 1 doh.example.net (
+                        ipv4hint=x.y.z.w
+                        dohuri=https://doh.example.net/dns-query )
+~~~
+
+#### SVCB Records for Equivalent Encrypted Resolvers using DoT
+
+The following example shows an Equivalent Encrypted Resolver using DoT, as returned by a query
+for an SVCB record for "dns://resolver.arpa":
+
+~~~
+   _dns.resolver.arpa  7200  IN SVCB 1 dot.example.net (
+                        ipv4hint=x.y.z.w
+                        dothostname=dot.example.net )
+~~~
+
+## Opportunistic Discovery from Unencrypted Resolvers
+
+There are situations where authenticated discovery of encrypted DNS configuration over
+unencrypted DNS is not possible. This includes unencrypted resolvers on non-public IP
+addresses whose identity cannot be confirmed using PKI.
+
+Clients who wish to attempt opportunistic DNS encryption MUST try authenticated
+discovery first if the Unencrypted Resolver in use has a public IP address. If that
+fails or the Unencrypted Resolver does not have a public IP address, the client MAY
+attempt opportunistic encryption as defined in Section 4.1 of {{!RFC7858}} to the same
+IP address without validating the resolver identity.
+
+### Opportunistic Discovery Cannot Change IP Addresses
+
+Opportunistic discovery of DNS encryption MUST NOT identify a different IP address from
+the Unencrypted Resolver's IP address. This is why there are no mechanisms defined for
+Unencrypted Resolvers to advertise a different IP address unless it can be authenticated.
+
+The reasoning behind this is to ensure the threat model for opportunistic encryption is
+not weaker than simply continuing to use unencrypted DNS. If the IP address of the
+Unencrypted Resolver was acquired by the DNS client securely such as by manual configuration,
+allowing on-path attackers an opportunity to change the destination IP address for an
+encrypted connection would worsen the security of the client. 
+
+
+## Encrypted Resolvers advertising Equivalent Encrypted Resolvers
+
+A DNS client may want to discover other DNS encryption transports supported by a given
+Encrypted Resolver. This can be accomplished by using the same mechanism but sending an SVCB
+query to the name of the Encrypted Resolver instead of the resolver.arpa SUDN.
+
+This enables a resolver to modify its answer to match the name being queried. It also
+enables resolvers to recursively answer the query if it is for a name it is not authoritative
+for. An example where this would be useful is when a client has DoH and DoT configuration
+for resolver Foo but only DoH configuration for resolver Bar. If DoH is being blocked on
```

Comment:
Maybe use an `example.com` thing instead of Foo?

Reply:
Good call.